### PR TITLE
chore(Charts): clean up a11y violations in docs

### LINKED ISSUE #7675: Fix charts docs a11y violations
-  Duplicate IDs (probably shared between examples)
   -  Stack chart
   -  Tooltip chart
   -  Scatter chart
   -  Line chart
   -  Donut utilization chart
   -  Bullet chart
   -  Area chart
   -  Legend chart
   -  Bar chart
   -  Pie chart
   -  Donut chart
-  SVGs need alt text
   -  Tooltip chart
   -  Sparkline chart
   -  Bullet chart
-  Legend chart examples - interactive controls cannot be nested

### PR DESCRIPTION
**Pull request by nicolethoen**
**What**: Closes #7675 

Duplicate IDs in axis labels and/or legend labels
- Stack chart
- Tooltip chart
- Scatter chart
- Line chart
- Bullet chart
- Area chart
- Legend chart
- Bar chart
- Pie chart
- Donut chart

Duplicate id due to reused example names:
- Donut utilization chart

SVGs need accessible text
- Tooltip chart
- Sparkline chart
- Bullet chart

Legend chart examples - interactive controls cannot be nested. Fixed this by updating the ChartContainer's rendered svg's role from 'img' to 'group'

### CODE REVIEW

**Review by dlabrecq — CHANGES_REQUESTED** [review]

File: packages/react-charts/src/components/ChartLegend/examples/ChartLegend.md [hunk 1]
```diff
@@ -680,7 +680,7 @@ class LegendLinkPieChart extends React.Component {
         <Chart
           ariaDesc="Average number of pets"
           ariaTitle="Line chart example"
-          containerComponent={<ChartVoronoiContainer labels={({ datum }) => `${datum.name}: ${datum.y}`} constrainToVisibleArea />}
+          containerComponent={<ChartContainer role="group" labels={({ datum }) => `${datum.name}: ${datum.y}`} constrainToVisibleArea />}
```
Comment (dlabrecq) [inline packages/react-charts/src/components/ChartLegend/examples/ChartLegend.md:683]: This needs to be `ChartVoronoiContainer` to show tooltip values related to x and y coordinates.

In addition, I envisioned adding `role="group"` as a default.

File: packages/react-charts/src/components/ChartAxis/ChartAxis.tsx [hunk 2]
```diff
@@ -462,7 +470,14 @@ export const ChartAxis: React.FunctionComponent<ChartAxisProps> = ({
   });
 
   // Note: containerComponent is required for theme
-  return <VictoryAxis containerComponent={container} theme={showGrid ? getAxisTheme(themeColor) : theme} {...rest} />;
+  return (
+    <VictoryAxis
+      containerComponent={container}
+      theme={showGrid ? getAxisTheme(themeColor) : theme}
+      tickLabelComponent={<VictoryLabel id={axisLabelId} />}
```
Comment (dlabrecq) [inline packages/react-charts/src/components/ChartAxis/ChartAxis.tsx:477]: We would want to use our own ChartLabel here. 

We also need to clone the given `tickLabelComponent` here, so we don't override a devs component properties.

**Review by dlabrecq — CHANGES_REQUESTED** [review]

File: packages/react-charts/src/components/ChartPoint/ChartPoint.test.tsx [hunk 3]
```diff
@@ -2,10 +2,13 @@ import * as React from 'react';
 import { render } from '@testing-library/react';
 import { ChartLegend } from '../ChartLegend';
 import { ChartPoint } from './ChartPoint';
+import { ChartLabel } from '../ChartLabel';
 
 Object.values([true, false]).forEach(() => {
   test('ChartPoint', () => {
-    const { asFragment } = render(<ChartLegend dataComponent={<ChartPoint />} />);
+    const { asFragment } = render(
+      <ChartLegend dataComponent={<ChartPoint />} titleComponent={<ChartLabel id="test" />} />
```
Comment (dlabrecq) [inline packages/react-charts/src/components/ChartPoint/ChartPoint.test.tsx:10]: This is another case where we must clone the `titleComponent`, so we don't override the dev's given properties

File: packages/react-charts/src/components/ChartPoint/ChartPoint.test.tsx [hunk 4]
```diff
@@ -17,6 +20,7 @@ test('renders component data', () => {
       title="Average number of pets"
       height={50}
       width={200}
+      titleComponent={<ChartLabel id="test2" />}
```
Comment (dlabrecq) [inline packages/react-charts/src/components/ChartPoint/ChartPoint.test.tsx:23]: Same here. We must clone the `titleComponent`, so we don't override the dev's given properties

**Review by dlabrecq — CHANGES_REQUESTED** [review]

File: packages/react-charts/src/components/ChartLegend/examples/ChartLegend.md (hunk 1 above)
Comment (dlabrecq) [reply packages/react-charts/src/components/ChartLegend/examples/ChartLegend.md:683]: It appears that Victory uses role="img" as the default to describe the SVG element

**Review by thatblindgeye — APPROVED** [review]
Looks good! I did note an issue I noticed below, but because what I noted *might* involve more investigation/possible larger changes, I wouldn't block over it.

File: packages/react-charts/src/components/ChartLegend/examples/ChartLegend.md [hunk 5]
```diff
@@ -680,7 +680,7 @@ class LegendLinkPieChart extends React.Component {
         <Chart
           ariaDesc="Average number of pets"
           ariaTitle="Line chart example"
-          containerComponent={<ChartVoronoiContainer labels={({ datum }) => `${datum.name}: ${datum.y}`} constrainToVisibleArea />}
+          containerComponent={<ChartVoronoiContainer role="figure" labels={({ datum }) => `${datum.name}: ${datum.y}`} constrainToVisibleArea />}
```
[the remark this answers is not in the document]
Comment (nicolethoen) [reply packages/react-charts/src/components/ChartLegend/examples/ChartLegend.md:683]: hm - yeah, @thatblindgeye do you mind opening an issue to address this? It's helpful and overall, charts could use a bit more a11y evaluation. I think it's just a bit out of scope here.

Once you do that, I think you or @dlabrecq could probably merge this

[the remark this answers is not in the document]
Comment (dlabrecq) [reply packages/react-charts/src/components/ChartLegend/examples/ChartLegend.md:683]: We have a couple a11y investigation issues open already for charts. In fact, our team was responsible for exposing the role attributes in the Victory upstream. There is some follow up work to add default roles, but SVGs will always have a11y quirks.

[the remark this answers is not in the document]
Comment (dlabrecq) [reply packages/react-charts/src/components/ChartLegend/examples/ChartLegend.md:683]: Not sure if this is helpful, but Christy suggested we might use `role=table`.
See https://github.com/patternfly/patternfly-react/issues/4135#issuecomment-625441882

VO announces the `desc` and `title` for `<svg role="table"`, but it identifies the chart as a table with zero rows. Screen readers won't read nested elements under `<svg role="img"`.

[the remark this answers is not in the document]
Comment (dlabrecq) [reply packages/react-charts/src/components/ChartLegend/examples/ChartLegend.md:683]: FYI, I tried using `containerId`, but didn't seem to work with `aria-describedby` and `aria-labelledby`. Using `aria-label` seemed to work a little better, but VO didn't announce `aria-details`.

```
<figure aria-describedby="chart" aria-labelledby="chart">
  <Chart
    ariaDesc="This is a descripton"
    ariaTitle="This is a title"
    containerComponent={<ChartContainer containerId="chart" role="figure" />}
  >
  ...
</figure
```

[the remark this answers is not in the document]
Comment (dlabrecq) [reply packages/react-charts/src/components/ChartLegend/examples/ChartLegend.md:683]: @nicolethoen This seems to work fairly well.

Screen Shot 2022-07-18 at 7 56 56 PM

Note that I'm using `containerId="chart"` with `ChartVoronoiContainer`. This allows me to add aria-`describedby="chart-desc"` and `aria-labelledby="chart-title"` to the outermost `div` tag.

VO now reads this chart's description and title, similar to `role=img`. Still using `role="figure"` with `ChartVoronoiContainer`, so VO can read each link.

VO recording:
https://user-images.githubusercontent.com/17481322/179636793-4daa6a16-615a-4163-adc8-5af8aff109b1.mov

[the remark this answers is not in the document]
Comment (nicolethoen) [reply packages/react-charts/src/components/ChartLegend/examples/ChartLegend.md:683]: nice!

[the remark this answers is not in the document]
Comment (thatblindgeye) [reply packages/react-charts/src/components/ChartLegend/examples/ChartLegend.md:683]: >nice!

Echoing this. The updates made by Nicole + the tweaks shown in the above comment make for a definite improvement of the current behavior.

**Review by dlabrecq — APPROVED** [review]
LGTM

**Review by thatblindgeye — APPROVED** [review]

**Pull request merged**